compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/indy/LambdaMetafactoryArguments.kt

```diff
@@ -77,20 +76,25 @@ internal class LambdaMetafactoryArgumentsBuilder(
         if (samClass.requiresDelegationToDefaultImpls())
             return null
 
+        // TODO in cases where LambdaMetafactory is unusable directly due to problems with the implementation function,
+        //      we can still avoid generating an entire class by converting the function reference into a lambda first;
+        //      see `InlineCallableReferenceToLambdaPhase`
```

Comment:
Yes, I've thought about that (and did some performance measurements on artificial projects). The benefit of creating a function reference with `invokedynamic` instead of an explicit class is rather marginal, and, in fact, it might be even a disadvantage in some very specific corner cases, since you effectively move some work from compile time to execution time, while the rest (loading a class, creating an instance, invoking implementation method) remains the same. I see no actual reason to take extra actions (like creating additional methods) just for the sake of using `invokedynamic`.